src/user/user.controller.ts

```diff
@@ -18,17 +19,16 @@ export class UserController {
   async registerUser(@Body() createUserDto: CreateUserDto) {
     return this.userService.registerUser(createUserDto);
   }
+  @Get()
+  async getAll(@Query('role') role: string) {
```

Comment:
The user `role` you need is obtained after decoding the JWT token, not from the request query parameters.
user role must be available `req.user.role`